group-coordinator/src/main/java/org/apache/kafka/coordinator/group/OffsetConfig.java

```diff
@@ -20,16 +20,51 @@
 import org.apache.kafka.common.record.CompressionType;
 
 public class OffsetConfig {
```

Comment:
@OmniaGM Thanks for working on this. I wonder whether we could merge `OffsetConfig` into `GroupCoordinatorConfig` in order to have one config for the entire package. It would bring us closer to the goal described in https://issues.apache.org/jira/browse/KAFKA-15089. Would it be possible?

Reply:
Good call, this actually will make it simplier. Will update the PR to reflict this

Reply:
I'll move all props, defaults and docs into `GroupCoordinatorConfig` and move the constructor for `OffsetConfig` as static class within `GroupCoordinatorConfig` as `GroupMetadataManager` which mainly cares about `OffsetConfig` doesn't need the rest of the configs in `GroupCoordinatorConfig`

Reply:
I pushed the changes let me know if it makes sense or if you have other suggestions/feedback.

Reply:
It is also bit odd to have a class  only for the constructor that do nothing but grouping. But I can see that split it out might be better as it seems like we have `GroupConfig` in scala that will get migrated to java at some point so wouldn't make sense to have the `GroupCoordinatorConfig` grow out of hand.

Reply:
`OffsetConfig` is only used by Scala code so it will disappear when we remove it. The `GroupConfig` in Scala may never be migrated to Java as `GroupCoordinatorConfig` already contains everything, I think. I wonder if we could actually replace `OffsetConfig` by an interface and make `GroupCoordinatorConfig` implements it. Then, we could pass `GroupCoordinatorConfig` to the old code too. I am not sure if this is feasible though.

Reply:
> OffsetConfig is only used by Scala code so it will disappear when we remove it

or we can move `OffsetConfig` to `kafka.coordinator.group` since it is used by `kafka.coordinator.group.GroupCoordinator` and `kafka.coordinator.group.GroupMetadataManager`.

Reply:
I moved it to `kafka.coordinator.group` for now we can have a followup for
[KAFKA-15089](https://github.com/apache/kafka/pull/15684/commits/581e9a88089aa9881ec28de75877350e17711b3e) to look into removing this class later.

Reply:
I would suggest that we wait until KafkaConfig is fully migrated out and then we can change the constructor of `GroupCoordinatorConfig` to accept KafkaConfig and it extract any needed grouping out KafkaConfig definition. WDYT? This might be the easiest way

Reply:
> then we can change the constructor of GroupCoordinatorConfig to accept KafkaConfig 

that makes sense to me. A unified way to generate those config class can make consistent behavior.

> I moved it to kafka.coordinator.group 

I assume the package you mentioned is in core module, but I'm ok with your approach since it can avoid rewriting java code back to scala code.

Reply:
sorry I meant `org.apache.kafka.coordinator.group`

Reply:
I updated the pr now. And rebased from trunk as well